clang/lib/CodeGen/CGCall.cpp

```diff
@@ -5612,6 +5612,10 @@ RValue CodeGenFunction::EmitCall(const CGFunctionInfo &CallInfo,
                               BundleList);
     EmitBlock(Cont);
   }
+  if (CI->getCalledFunction() && CI->getCalledFunction()->hasName() &&
+      CI->getCalledFunction()->getName().startswith("_Z4sqrt")) {
```

Comment:
This isn't a specific enough filter. Exactly match the full name for all the types?

Reply:
Maybe it's good enough, I guess the Z4 filters out other prefixes. Not sure about arbitrary custom ext_vector_type

Reply:
Using a custom `ext_vector_type` in OpenCL feels like an undefined behaviour. I don't see anything about it in the specification.
Am I missing something?

Reply:
@arsenm ?

Reply:
ext_vector_type is an implementation detail. The standardized vector types just happen to be typedefs of ext_vector_type.  It's not undefined behavior, it's just something out of spec you can do